ra_aid/agent_utils.py

```
@@ -56,58 +64,210 @@
+        provider_tokens = models_tokens.get(provider, {})
+        max_input_tokens = provider_tokens.get(normalized_name, None)
+        if max_input_tokens:
+            logger.debug(
+                f"Found token limit for {provider}/{model_name}: {max_input_tokens}"
+            )
+        else:
+            logger.debug(f"Could not find token limit for {provider}/{model_name}")
+
+        return max_input_tokens
+
+    except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)